### PR TITLE
WIP: Doing Docs at Crate.io

### PR DESCRIPTION
**Pull request by norosa**
WIP

summary context for now:

- was partway through migrating old wiki content to new `dev/docs` dir. realized I could really do with automated link checks for the document I was writing. don't want to re-invent the wheel. may as well move into sphinx and take advantage of our existing infra

- while I'm at it, why not move existing RST files under `style` into sphinx for the same reason (haven't done this yet. also haven't moved the wiki exports I already added to `devs/docs`)

- named the project "doing the docs at crate.io" for now. will be sparse at first. but a good replacement for our old private wiki and something we can grow over time

- expect I can find some extra bits and bobs to sprinkle in before merging

### CODE REVIEW

**Comment by norosa** [comment]
checkpoint commit because I want to add a new module to the crate-docs-theme and I would like to reference my WIP to justify my change. PR incoming shortly

File: docs/index.rst [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+======================
 
-This documentation project is for demo purposes only.
+The `Crate.io Docs`_ project contains the :ref:`common build system
+<common-build>`, :ref:`QA tools <qa>`, :ref:`style guides <style>`, and other
+meta miscellanies for doing docs at `Crate.io`_.
+
+.. NOTE::
+
+    We have designed this project so that it also function as a reference
```
Comment (amotl) [inline docs/index.rst:13]:
```suggestion
    We have designed this project so that it also functions as a reference
```

File: docs/linking.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
+=======
+Linking
+=======
+
+Short Links
+===========
+
+We have configured a number of short links (i.e., URLs) for the docs. These
+URLs can be used when the naked URL is going to be visible, for improved
+readability and use of character space..
```
Comment (amotl) [inline docs/linking.rst:10]:
```suggestion
readability and use of character space.
```

File: docs/linking.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- crate.io/cloud/howtos/
+- crate.io/cloud/reference/
+- crate.io/cloud/croud/
+
+
+Latest versions
+===============
+
+The short links always redirect to the latest version of the documentation. So,
+for ecample, in the case of the CrateDB Reference,
```
Comment (amotl) [inline docs/linking.rst:34]:
```suggestion
for example, in the case of the CrateDB reference documentation,
```

File: docs/linking.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+``https://crate.io/docs/crate/reference/en/latest/``.
+
+In turn, the ``/en/latest/`` URL will redirect to the latest stable version of
+the docs. For example, at the time of writing, that means the final URL is
+``https://crate.io/docs/crate/reference/en/4.5/``.
+
+The ``latest`` version *always* redirects to the latest stable version, and as
+such, you will never find it visible in the URL bar in your browser.
+
+IF you copy the URL from your browser when you want to link to a page in the
```
Comment (amotl) [inline docs/linking.rst:45]:
```suggestion
If you copy the URL from your browser when you want to link to a page in the
```

File: docs/linking.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+documentation, the URL effectively pins the documentation to the current stable
+version. This is fine for in most cases (e.g., for most day-to-day use by
```
Comment (amotl) [inline docs/linking.rst:47]:
```suggestion
version. This is fine in most cases (e.g., for most day-to-day use by
```

File: docs/linking.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+readers).
+
+However, if you are creating more a more permanent cross-reference, in most
+cases, you should link using a URL that uses the `latest` version. Doing so
```
Comment (amotl) [inline docs/linking.rst:51]:
```suggestion
cases, you should link using a URL that uses the ``latest`` version. Doing so
```

File: docs/linking.rst [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+will ensure that the link you created continues to take the reader to the
+latest version of that page forever.
+
+Example
+-------
+
+Let's say you are adding content to the `inserts performance` section of the CrateDB How-To Guides.
```
Comment (amotl) [inline docs/linking.rst:58]:
```suggestion
Let's say you are adding content to the `inserts performance`_ section of the CrateDB How-To Guides.
```

File: docs/projects/post-release.rst [hunk 8]
```diff
@@ -0,0 +1,173 @@
+.. _projects-post-release:
+
+================================
+Post-Release Docs Update Process
+================================
+
+When `CrateDB`_ makes a new stable release, the CrateDB release manager will
```
Comment (amotl) [inline docs/projects/post-release.rst:7]:
```suggestion
When CrateDB_ makes a new stable release, the CrateDB release manager will
```

File: docs/projects/post-release.rst [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+ping a tech writer to update the docs.
+
+The `CrateDB docs` make use of a special ``latest`` version in the `Read The
```
Comment (amotl) [inline docs/projects/post-release.rst:10]:
```suggestion
The `CrateDB docs`_ make use of a special ``latest`` version in the `Read The
```

File: docs/projects/post-release.rst [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+  rewrite ^/docs/crate/reference/en/latest/(.*)$ /docs/crate/reference/en/4.2/$1 permanent;
+  ```
+
+  This redirect allows us to link to pages in the documentation corresponding to the latest stable release version by using ``/latest/`` as the version parameter in URLs.
+
+  All links to the documentation should use ``/latest/`` as the version parameter unless there is a good reason to link to a specific version number.
+
+- Skip to the *Active versions* section
+
+- Each active release line should have a corresponding entry this section, like so:
```
Comment (amotl) [inline docs/projects/post-release.rst:86]: Not sure about this?

```suggestion
- Each active release line should have a corresponding entry, like so:
```

File: docs/projects/post-release.rst [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+
+  ```
+  rewrite ^/docs/crate/reference/en/4\.(?!1/)[^/]*/(.*)$ /docs/crate/reference/en/4.1/$1 permanent;
+  ```
+
+  This redirect checks for any 4.x version that doesn't match the latest feature release (e.g., ``4.1`) for that release line and redirects the request to the latest feature release version (e.g., ``4.1``). In this example, this redirect forces all 4.x versions to resolve to 4.1.
```
Comment (amotl) [inline docs/projects/post-release.rst:92]:
```suggestion
  This redirect checks for any 4.x version that doesn't match the latest feature release (e.g., ``4.1``) for that release line and redirects the request to the latest feature release version (e.g., ``4.1``). In this example, this redirect forces all 4.x versions to resolve to 4.1.
```

**Review by amotl — CHANGES_REQUESTED** [review]
Hi Naomi,

thank you for all of those refinements. While I recognize this patch is still in draft mode, I quickly took a look and added some comments as suggestions. I believe most of them are merely typos.

With kind regards,
Andreas.

File: docs/projects/post-release.rst (hunk 11 above)
Comment (norosa) [reply docs/projects/post-release.rst:92]: nice catch

**Comment by norosa** [comment]
@amotl hehe thanks for the (very) early review. pushed some fixups. only one I didn't apply. left a comment. more to come soon

File: docs/projects/post-release.rst (hunk 8 above)
Comment (amotl) [reply docs/projects/post-release.rst:7]: All right, it's fine!

File: docs/projects/post-release.rst [hunk 12]
```diff
@@ -0,0 +1,173 @@
+.. _projects-post-release:
+
+================================
+Post-Release Docs Update Process
+================================
```
Comment (amotl) [inline docs/projects/post-release.rst:5]: Hi Naomi,

when thinking about this again, I believe we should not publish that kind of specific information here. Can you remove this document `post-release.rst` completely?

With kind regards,
Andreas.

**Review by amotl — CHANGES_REQUESTED** [review]
Hi Naomi,

there might be a chance to get this patch merged? However, I requested a specific change about removing information which I believe is too specific for this general/public repository.

With kind regards,
Andreas.

**Pull request closed without merging**